README.md

```diff
@@ -459,6 +528,93 @@ with `ExecMsg/QueryMsg` - the former is generated only for contract, not for int
 and is not meant to use to send messages to the contract - their purpose is for proper
 messages dispatching only, and should not be used besides the entry points.
 
+## Query helpers
+
+To make querying more user friendly `Sylvia` generates `BoundQuerier` and `Remote` helpers.
+The latter is meant to store the address of some remote contract. It's generated implementation
+looks like this:
+
+```rust
+#[derive(sylvia::serde::Serialize, sylvia::serde::Deserialize)]
+pub struct Remote<'a>(std::borrow::Cow<'a, sylvia::cw_std::Addr>);
+
+impl Remote<'_> {
+    pub fn querier<'a, C: sylvia::cw_std::CustomQuery>(
+        &'a self,
+        querier: &'a sylvia::cw_std::QuerierWrapper<'a, C>,
+    ) -> BoundQuerier<'a, C> {
+        BoundQuerier {
+            contract: &self.0,
+            querier,
+        }
+    }
+}
+```
+
+It has a single method implemented called querier which returns the `BoundQuerier` for the stored
+address.
+
+```rust
+
+pub struct BoundQuerier<'a, C: sylvia::cw_std::CustomQuery> {
+    contract: &'a sylvia::cw_std::Addr,
+    querier: &'a sylvia::cw_std::QuerierWrapper<'a, C>,
+}
+
+impl<'a, C: sylvia::cw_std::CustomQuery> Querier for BoundQuerier<'a, C> {
+    fn counter(&self) -> Result<CounterResp, sylvia::cw_std::StdError> {
+        let query = QueryMsg::counter();
+        self.querier.query_wasm_smart(self.contract, &query)
+    }
+}
+
+pub trait Querier {
+    fn counter(&self) -> Result<CounterResp, sylvia::cw_std::StdError>;
+}
+```
+
+For each query method in the contract `Sylvia` will implement via generated `Querier` trait
+method for more user friendly querying.
+
+Let's modify the query from the previous paragraph. Currently it will look as follows:
+
+```rust
+let is_member = Remote::new(remote_addr)
+    .querier(&ctx.deps.querier)
+    .is_member(addr)?;
+```
+
+Your contract might be implemented such it will be communicating with some other contract regularly.
+In such case you might want to store it as a field in your Contract:
+
+```rust
+pub struct MyContract<'a> {
+    counter: Item<'a, u64>,
+    members: Map<'a, &'a Addr, Empty>,
+    // Added
+    remote: Item<'a, Remote<'a>>,
```

Comment:
No, that is wrong. This is not a proper lifetime, `'a` is a lifetime for the name of an item; here we can store only the `'static` remote. If it works, it works by pure accident.